numba/core/typing/randomdecl.py

```diff
@@ -228,6 +228,36 @@ def typer(shape, scale=None, size=None):
             return self.array_typer(size)(shape, scale)
         return typer
 
+@glue_typing(np.random.uniform, typing_key="np.random.uniform")
+class Numpy_uniform(ConcreteRandomTemplate):
+    cases = [signature(tp, tp, tp) for tp in _float_types]
+
+    def generic(self):
+        def typer(low, high, size=None):
+            return self.array_typer(size)(low, high)
+        return typer
+
+
+@glue_typing(np.random.gumbel, typing_key="np.random.gumbel")
+class Numpy_gumbel(ConcreteRandomTemplate):
+    cases = [signature(tp, tp, tp) for tp in _float_types]
+
+    def generic(self):
+        def typer(loc, scale, size=None):
+            return self.array_typer(size)(loc, scale)
+        return typer
+
+
+@glue_typing(np.random.wald, typing_key="np.random.wald")
+class Numpy_wald(ConcreteRandomTemplate):
+    cases = [signature(tp, tp, tp) for tp in _float_types]
+
+    def generic(self):
+        def typer(mean, scale, size=None):
+            return self.array_typer(size)(mean, scale)
+        return typer
```

Comment:
Numba is transitioning to use the [high-level API](https://numba.pydata.org/numba-doc/latest/extending/high-level.html) for implementing NumPy functions. Once the transition is done, the `glue_*` machinery will not be necessary anymore. @apmasell is working on migrating the `np.random.*` module to use the high-level API on https://github.com/numba/numba/pull/8061. With that said, I wonder if this change worth or if it better to just try using the high level API.

Reply:
Hi @guilhermeleobas, thanks for your review. I took a look at [#8061](https://github.com/numba/numba/pull/8061), and yes, the implementations using new API solve #8161 perfectly fine. I see that some distributions are not yet implemented, and for those ones #8161 remains an issue (8161 specifically concerns np.random.uniform, and that one is not yet converted to the new API).  On the other hand, Gumbel and Wald are already implemented, and my unit tests pass on that branch. The question is: how can I include the high level API without plagiarizing the work done by @apmasell ?

Reply:
Hi @bszollosinagy, to avoid re-doing @apmasell work. I think it is better to just wait his PR to land. As for `np.random.uniform`, if you can, you can do the migration for this particular function in a new PR.

Reply:
Hopefully my changes will land soon, but there will be another round of changes to push the migrate the random functions that generate IR directly.

Reply:
Okay, I'll hold on, and when #8061 lands, then I'll merge the main into this PR and adapt it.

Reply:
Okay, I see that #8061 is merged. So now I take it by merging main into my branch, and solve the merge conflicts by dropping the old glue code.  For the uniform distrib I keep the glue code, because that is not yet implemented in the new style.

Reply:
The new unit tests pass (those that test that keyword based calling works as intended). 
So the issue is solved for the uniform distribution (which was the main task for #8161 ). In total, these distributions now work well with keyword arguments: (Wald and Gumbel were done by @apmasell in PR #8061  by using override instead of glue, which implicitly solved the issue)

- np.random.uniform
- np.random.gumbel
- np.random.wald

This PR is now ready for review